odxtools/admindata.py

```diff
@@ -46,28 +66,133 @@ class Modification:
     change: Optional[str] = None
     reason: Optional[str] = None
 
+    @staticmethod
+    def from_et(et_element: ElementTree.Element,
+                doc_frags: List[OdxDocFragment]) \
+            -> "Modification":
+        change = et_element.findtext("CHANGE")
+        reason = et_element.findtext("REASON")
+
+        return Modification(change=change,
+                            reason=reason)
+
+@dataclass()
+class CompanyRevisionInfo:
+    company_data_ref: OdxLinkRef
+    revision_label: Optional[str] = None
+    state: Optional[str] = None
+
+    _company_data: Optional[CompanyData] = None
+    @property
+    def company_data(self) -> CompanyData:
+        assert self._company_data is not None
+        return self._company_data
+
+    @staticmethod
+    def from_et(et_element: ElementTree.Element,
+                doc_frags: List[OdxDocFragment]) \
+            -> "CompanyRevisionInfo":
+
+        company_data_ref = OdxLinkRef.from_et(et_element.find("COMPANY-DATA-REF"),
+                                              doc_frags)
+        assert company_data_ref is not None
```

Comment:
required in XSD?

Reply:
no. thanks for the catch!

Reply:
ok, I misread the class which this was about. for COMPANY-REVISION-INFO the COMPANY-DATA-REF tag is indeed mandatory...